src/snowcli/cli/nativeapp/manager.py

```diff
@@ -35,3 +60,101 @@ def deploy_root(self) -> Path:
+                encryption = (TYPE = 'SNOWFLAKE_SSE');"""
+        self.execute_query_with_role(create_stage_if_not_exists_stmt, role)
+
+        # Perform a diff operation and display results to the user for informational purposes
+        log.info(
+            f"Performing a diff between the Snowflake stage and your local deploy_root {deploy_root} directory."
+        )
+        diff: DiffResult = stage_diff(
+            Path(deploy_root), f"{app_pkg}.{schema_and_stage}"
+        )
+        log.info("Listing results of diff:")
+        print_result(
+            ObjectResult(
+                {
+                    "only on local": ", ".join(diff.only_local),
+                    "only on stage": ", ".join(diff.only_on_stage),
+                    "modified/unknown": ", ".join(diff.different),
+                    "identical": ", ".join(diff.identical),
+                }
+            )
+        )
+
+        # Upload diff-ed files to app pkg stage
+        log.info(
+            f"Uploading diff-ed files from your local {deploy_root} directory to the Snowflake stage."
+        )
+        sync_local_diff_with_stage(
+            role=role, diff_result=diff, stage_path=f"{app_pkg}.{schema_and_stage}"
+        )
+
+    def app_run(self) -> SnowflakeCursor:
+
+        # It is possible that snowflake.yml does not contain some of the optional fields,
+        # and there is no snowflake.local.yml present in the search_path.
+        # In this case, we need to use a generated local yml which provides those default values.
+        generated_local_yml_overrides = generate_local_override_yml(
+            self.definition_manager.project_definition
+        )
+
+        app_pkg = get_required_field_from_definition(
+            "package", "name", self.definition, generated_local_yml_overrides
+        )
+
+        app_pkg_role = get_required_field_from_definition(
+            "package", "role", self.definition, generated_local_yml_overrides
+        )
+
+        show_app_cursor = self.execute_query_with_role(
+            f"show application packages like {app_pkg}", app_pkg_role
+        )
+
+        if show_app_cursor.rowcount == 0:
+            # Create an app pkg, with distribution = internal to avoid triggering security scan
+            log.info(f"Creating new application package {app_pkg} in account.")
+            self.execute_query_with_role(
+                f"""
+                create application package {app_pkg}
+                    comment = {SPECIAL_COMMENT}
+                    distribution = internal
+            """,
+                app_pkg_role,
+            )
+        else:
+            # There can only be one possible pre-existing app pkg with the same name
+            show_app_row = show_app_cursor.fetchone()
+            if isinstance(show_app_row, dict):
+                row_comment = show_app_row[COMMENT_COL]
+            else:  # is list otherwise
+                row_comment = show_app_row[6]
```

Comment:
How this can happen?

Reply:
This is a good question. Actually it was something we were discussing earlier; not sure how exactly to know if a query will return dicts or tuples. Do you know?

Reply:
I saw a list implementation, in streamlit's manager under `deploy`:
```
base_url = query_result.fetchone()[0]
``` 
but the return type is a union of dict and list, so I added a case for either return type.